hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/ContractServiceModule.java

```diff
@@ -58,16 +61,19 @@ public interface ServiceModule {
 
     @Binds
     @IntoMap
+    @Singleton
```

Comment:
In this case we do need the `@Singleton` annotation, because this binding is part of a "multibound collection"; that is, a
```
Map<HederaEvmVersion, TransactionProcessor> processors
```

The object that's created by Dagger when you request a multibound set or map will have the scope of the component that you request it from.

Since we're "requesting" this `Map<HederaEvmVersion, TransactionProcessor>` [here](https://github.com/hashgraph/hedera-services/blob/06487-scope-impl/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/ContextTransactionProcessor.java#L66), inside a `@TransactionScope` component, without the `@Singleton` annotation here, each of its three `TransactionProcessor`'s would be re-created for every transaction and new `TransactionComponent`.

(Arguably we should avoid re-creating this `Map` on every transaction by injecting it into e.g. `ContractCreateHandler` and then adding a `@BindsInstance` parameter to the `TransactionComponent.Factory.create()` method...but this seems like premature optimization. 😁)

Reply:
I have to think about this, and seriously start playing with Dagger.  It sure is convenient - I totally get that - but at the same time there's a bit of "action at a distance" going on that's kind of spooky, because invisible.